### PR TITLE
Sneak peek: Recipe app

### PR DESCRIPTION
**Pull request by ndikumanaisaie**
## In this milestone we @ndikumanaisaie @mkaya13 @Linushaddai99 covered the following:

- Followed the layout of the wireframes provided. 
- Login page and registration page build with Devise.
- Food list:
  - display a list of food added by the logged-in user as in the wireframe.
  - lead to a form that allows users to add new food.
- Recipes list:
  - display a list of recipes created by the logged-in user as in the wireframe.
  - lead to recipe details.
  - If the user is the owner of the recipe, the user can delete it.
- Public recipe list:
- display a list of all public recipes ordered by the newest as in the wireframe.
  - lead to recipe details.
- Recipe details:
  - display a toggle button that allows for a recipe to be made public or private.
  - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
  - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- Created a navigation menu that allows users to open all of the pages you created.
- Wrote unit and integration tests.
- Inventories list:
  - display a list of inventories created by the logged-in user as in the wireframe.
  - lead to inventory details.
  - If the user is the owner of the inventory, she is allowed to delete it.
- Inventory details:
  - display the inventory details as in the wireframe.
  - lead to a form that allows users to add new food.
- Recipe details:
  - have a modal to choose an inventory to create a shopping list with, as in the wireframe.
  - Inventory shopping list:
  - show the list of food that is missing by comparing the food in the recipe with the food in the inventory.
  - count the total number of food items and the total price of the missing food.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,163 @@
...
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+    <!-- - [🚀 Live Demo](#live-demo) -->
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
+  <!-- - [Deployment](#triangular_flag_on_post-deployment) -->
```
Comment (arslanbisharat) [inline README.md:16]: - [x] There are no instructions on how to run the tests Please add that so once would know how to do it,

File: README.md [hunk 2]
```diff
@@ -1 +1,163 @@
...
+- To have [Ruby](https://www.ruby-lang.org/en/) and [Git](https://git-scm.com/) installed on your system.
+- To have [Node.js](https://nodejs.org/) installed on your system.
+
+After installing Ruby, run the following command to install rails.-
+
+```sh
+ gem install rails
+```
+
+### Setup
```
Comment (arslanbisharat) [inline README.md:71]: - [x] I am unable to open the app on the live server by using the instructions provided by you. Kindly provide proper instructions.

File: README.md [hunk 3]
```diff
@@ -1 +1,163 @@
...
+
+OR
+
+```sh
+  rails s
+```
+
+<!-- ### Run tests
+
+To run tests, run the following command:
```
Comment (arslanbisharat) [inline README.md]: - [x] Upon running tests, I am getting the following error.  Let's fix it: 
image

File: app/assets/stylesheets/recipeIndex.css [hunk 4]
```diff
@@ -0,0 +1,97 @@
+#recipes-holder {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-content: center;
+  width: 100%;
+  height: 100%;
+}
+
+#recipes {
```
Comment (arslanbisharat) [inline app/assets/stylesheets/recipeIndex.css]: - [x] Let's not use ids as selectors in your code as it makes the code dry.

File: app/controllers/foods_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+class FoodsController < ApplicationController
+  # before_action :set_food, only: %i[new show destroy]
+
+  # GET /foods or /foods.json
+  def index
+    # @user = current_user
+    # @foods = Food.where(user_id: @user.id)
```
Comment (arslanbisharat) [inline app/controllers/foods_controller.rb]: - [x] let's remove commended code as it makes code hard to read.

File: app/controllers/inventories_controller.rb [hunk 6]
```diff
@@ -0,0 +1,89 @@
+class InventoriesController < ApplicationController
+  before_action :set_inventory, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+
+  # GET /inventories or /inventories.json
+  def index
+    # @inventories = Inventory.all
```
Comment (arslanbisharat) [inline app/controllers/inventories_controller.rb]: - [x] let's remove commented code as it makes code hard to read.

File: app/controllers/inventories_controller.rb [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+    return unless current_user
+
+    return unless current_user.present?
+
+    # @inventories = Inventory.includes([:user]).where(user: current_user).order(created_at: :desc)
```
Comment (arslanbisharat) [inline app/controllers/inventories_controller.rb]: - [x] let's remove commented code as it makes code hard to read.

File: app/controllers/inventories_controller.rb [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    @inventories = Inventory.where(user: current_user).all
+
+    # @inventories = Inventory.includes(:user).find(params[:user_id])
+    # @posts = Post.where(author_id: params[:user_id])
```
Comment (arslanbisharat) [inline app/controllers/inventories_controller.rb]: - [x] let's remove commented code as it makes code hard to read.

File: app/controllers/inventories_controller.rb [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  end
+
+  # GET /inventories/1 or /inventories/1.json
+  def show
+    return unless current_user
+
+    @foods = Food.all
+    @meals = InventoryFood.where(inventory_id: params[:id])
+    # @inventory_food = InventoryFood.where(inventory_id: params[:inventory_id], id: params[:id])
```
Comment (arslanbisharat) [inline app/controllers/inventories_controller.rb]: - [x] let's remove commented code as it makes code hard to read.

File: app/controllers/inventories_controller.rb [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  end
+
+  # Only allow a list of trusted parameters through.
+  def inventory_params
+    #  params.fetch(:inventory, {name})
+    params.require(:inventory).permit(:name, :description).merge(user: current_user)
+  end
+end
+
+# params.require(:inventory).permit(:name, :description).merge(user: current_user)
```
Comment (arslanbisharat) [inline app/controllers/inventories_controller.rb]: - [x] let's remove commented code as it makes code hard to read.

File: app/controllers/inventory_foods_controller.rb [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+    #   if @inventory_food.save
+    #     format.html do
+    #       redirect_to inventory_food_url(@inventory_food), notice: 'Inventory food was successfully created.'
+    #     end
+    #     format.json { render :show, status: :created, location: @inventory_food }
+    #   else
+    #     format.html { render :new, status: :unprocessable_entity }
+    #     format.json { render json: @inventory_food.errors, status: :unprocessable_entity }
+    #   end
+    # end
```
Comment (arslanbisharat) [inline app/controllers/inventory_foods_controller.rb]: - [x] let's remove commented code as it makes code hard to read.

File: app/controllers/inventory_foods_controller.rb [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+  # Use callbacks to share common setup or constraints between actions.
+  # def set_inventory_food
+  #   @inventory_food = InventoryFood.find(params[:id])
+  # end
+
+  # Only allow a list of trusted parameters through.
+  # def inventory_food_params
+  #  params.fetch(:inventory_food, {})
+  # end
+
```
Comment (arslanbisharat) [inline app/controllers/inventory_foods_controller.rb]: - [x] let's remove commented code as it makes code hard to read.

File: app/models/user.rb [hunk 13]
```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (arslanbisharat) [inline app/models/user.rb]: - [x] let's remove commented code as it makes code hard to read.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @ndikumanaisaie, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (mkaya13) [reply README.md:16]: Following commit for the solution:

https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/76662abd273e22259bbcbd901be8bb545e01912d

File: README.md (hunk 2 above)
Comment (mkaya13) [reply README.md:71]: Following commit for the solution:
https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/44765d251db1f63cfb5b477f68e8a063c0a1d791

File: README.md (hunk 3 above)
Comment (mkaya13) [reply README.md]: I run the tests with rspec spec command and get the following results: 

image

File: app/assets/stylesheets/recipeIndex.css (hunk 4 above)
Comment (mkaya13) [reply app/assets/stylesheets/recipeIndex.css]: Following commit for the solution:

image

File: app/controllers/foods_controller.rb (hunk 5 above)
Comment (mkaya13) [reply app/controllers/foods_controller.rb]: Following commit for the solution:

https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/cb40a1da187ce86386e47dd0e40c0cf3546eeed9

File: app/controllers/inventories_controller.rb (hunk 6 above)
Comment (mkaya13) [reply app/controllers/inventories_controller.rb]: Following commit for the solution:

https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/cb40a1da187ce86386e47dd0e40c0cf3546eeed9

File: app/controllers/inventories_controller.rb (hunk 7 above)
Comment (mkaya13) [reply app/controllers/inventories_controller.rb]: Following commit for the solution:

https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/cb40a1da187ce86386e47dd0e40c0cf3546eeed9

File: app/controllers/inventories_controller.rb (hunk 8 above)
Comment (mkaya13) [reply app/controllers/inventories_controller.rb]: Following commits for the solutions:

- https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/cb40a1da187ce86386e47dd0e40c0cf3546eeed9
- https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/c31e47e2ed8ef63c5b6f1243739e10ede6b61d08

File: app/controllers/inventories_controller.rb (hunk 9 above)
Comment (mkaya13) [reply app/controllers/inventories_controller.rb]: Following commits for the solutions:

- https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/cb40a1da187ce86386e47dd0e40c0cf3546eeed9
- https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/c31e47e2ed8ef63c5b6f1243739e10ede6b61d08

File: app/controllers/inventories_controller.rb (hunk 10 above)
Comment (mkaya13) [reply app/controllers/inventories_controller.rb]: Following commits for the solutions:

- https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/cb40a1da187ce86386e47dd0e40c0cf3546eeed9
- https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/c31e47e2ed8ef63c5b6f1243739e10ede6b61d08

File: app/controllers/inventory_foods_controller.rb (hunk 11 above)
Comment (mkaya13) [reply app/controllers/inventory_foods_controller.rb]: Following commits for the solution:

- https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/6a573f9af5785fbcfe8398324428088a521b2dd9

File: app/controllers/inventory_foods_controller.rb (hunk 12 above)
Comment (mkaya13) [reply app/controllers/inventory_foods_controller.rb]: Following commits for the solution:

https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/6a573f9af5785fbcfe8398324428088a521b2dd9

File: app/models/user.rb (hunk 13 above)
Comment (mkaya13) [reply app/models/user.rb]: Following commit for the solution:

- https://github.com/Linushaddai99/Group-Recipe-app/pull/33/commits/fc628162f43d3f2c3eb1ad3554a036b4bfde7a22

File: README.md [hunk 14]
```diff
@@ -70,10 +70,14 @@ After installing Ruby, run the following command to install rails.-
...
 
-Clone this project on your local machine using the following command in your terminal:
+- Clone this project on your local machine using the following command in your terminal:
+
+- Make sure to add .env file and setup your .env file with your postgres username and password!
 
 ```sh
   git clone https://github.com/Linushaddai99/Group-Recipe-app.git
+  bundle install
+  rails s
```
Comment (pinkmoon25) [inline README.md]: - [ ] [OPTIONAL] A proper setup instruction should be 

1. clone the repo `clone link`
2. installation `bundle install`
3. create database `rails db:create`
4. run migrations` rails db:migrate`
5. start the server `rails s`

File: README.md [hunk 15]
```diff
@@ -102,16 +102,16 @@ OR
...
 Example command:
 
-```sh
-  gem install rails rspec
-  rails spec spec/models/
-``` -->
+```
+  bundle install
+  rspec spec
+```
```
Comment (pinkmoon25) [inline README.md]: - [ ] [OPTIONAL] Again for running tests we need to run the migrations and setup the database for test, which is done by running `rails db:migrate RAILS_ENV=test` then only `rspec spec` command will work properly

**Review by pinkmoon25 — APPROVED** [review]
Hi @mkaya13 @ndikumanaisaie @Linushaddai99 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Rubocop checks :heavy_check_mark: 
gitflow :heavy_check_mark: 
Pair-programming :heavy_check_mark: 
Foods, Recipes and Shopping list :heavy_check_mark: 
Rspecs :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**